File: back/schemas/loan.py
from pydantic import BaseModel
from datetime import date


class LoanBase(BaseModel):
    reader_id: int
    copy_id: int
    staff_id: int
    loan_date: date
    due_date: date
    status: str = "active"  # active, returned, overdue


class LoanCreate(BaseModel):
    reader_id: int
    copy_id: int
    staff_id: int
    due_date: date


class LoanUpdate(BaseModel):
    return_date: date | None = None
    status: str | None = None


class LoanResponse(LoanBase):
    id: int
    return_date: date | None = None

    model_config = {"from_attributes": True}


class LoanWithDetails(LoanResponse):
    book_title: str | None = None
    reader_name: str | None = None
    staff_name: str | None = None
    inventory_number: str | None = None

File: LibraSmartWPF/Models/Loan.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LibraSmartWPF.Models;

/// <summary>
/// Выдачи книг (активные займы)
/// </summary>
[Table("loans")]
public class Loan
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required]
    [Column("reader_id")]
    public int ReaderId { get; set; }

    [Required]
    [Column("copy_id")]
    public int CopyId { get; set; }

    [Required]
    [Column("staff_id")]
    public int StaffId { get; set; }

    [Required]
    [Column("loan_date")]
    public DateTime LoanDate { get; set; }

    [Required]
    [Column("due_date")]
    public DateTime DueDate { get; set; }

    [Column("return_date")]
    public DateTime? ReturnDate { get; set; }

    [Required]
    [MaxLength(20)]
    [Column("status")]
    public string Status { get; set; } = "active"; // active, returned, overdue

    // Связи
    [ForeignKey(nameof(ReaderId))]
    public virtual Reader Reader { get; set; } = null!;

    [ForeignKey(nameof(CopyId))]
    public virtual BookCopy Copy { get; set; } = null!;

    [ForeignKey(nameof(StaffId))]
    public virtual Staff Staff { get; set; } = null!;
}
